packages/mermaid/src/diagrams/sankey/sankeyRenderer.ts

```diff
@@ -0,0 +1,179 @@
+// @ts-nocheck TODO: fix file
+import { Diagram } from '../../Diagram.js';
+import * as configApi from '../../config.js';
+
+import {
+  select as d3select,
+  // @ts-ignore TODO: make proper import
+  scaleOrdinal as d3scaleOrdinal,
+  // @ts-ignore TODO: make proper import
+  schemeTableau10 as d3schemeTableau10,
+} from 'd3';
+
+import {
+  sankey as d3Sankey,
+  sankeyLinkHorizontal as d3SankeyLinkHorizontal,
+  sankeyLeft as d3SankeyLeft,
+  sankeyRight as d3SankeyRight,
+  sankeyCenter as d3SankeyCenter,
+  sankeyJustify as d3SankeyJustify,
+} from 'd3-sankey';
+import { configureSvgSize } from '../../setupGraphViewbox.js';
+import { Uid } from './sankeyUtils.js';
+
+/**
+ * Draws a sequenceDiagram in the tag with id: id based on the graph definition in text.
+ *
+ * @param text - The text of the diagram
+ * @param id - The id of the diagram which will be used as a DOM element id¨
+ * @param _version - Mermaid version from package.json
+ * @param diagObj - A standard diagram containing the db and the text and type etc of the diagram
+ */
+export const draw = function (text: string, id: string, _version: string, diagObj: Diagram): void {
+  // TODO: Figure out what is happening there
+  // The main thing is svg object that is a d3 wrapper for svg operations
+  //
+  const { securityLevel, sequence: conf } = configApi.getConfig();
+  let sandboxElement: any;
+  if (securityLevel === 'sandbox') {
+    sandboxElement = d3select('#i' + id);
+  }
+  let root = d3select('body');
+
+  if (securityLevel === 'sandbox' && sandboxElement) {
+    root = d3select(sandboxElement.nodes()[0].contentDocument.body);
+  }
+  const doc = securityLevel === 'sandbox' ? sandboxElement.nodes()[0].contentDocument : document;
+  const svg = securityLevel === 'sandbox' ? root.select(`[id="${id}"]`) : d3select(`[id="${id}"]`);
+
+  // Establish svg dimensions and get width and height
+  //
+  const elem = doc.getElementById(id);
+  const width = elem.parentElement.offsetWidth;
+  const height = 600;
+
+  // FIX: using max width prevents height from being set
+  configureSvgSize(svg, height, width, false);
+  // svg.attr('height', height); // that's why we need this line
+
+  // Prepare data for construction based on diagObj.db
+  // This must be a mutable object with 2 properties:
+  // `nodes` and `links`
+  //
+  //    let graph = {
+  //      "nodes": [
+  //        { "id": "Alice" },
+  //        { "id": "Bob" },
+  //        { "id": "Carol" }
+  //      ],
+  //      "links": [
+  //        { "source": "Alice", "target": "Bob", "value": 23 },
+  //        { "source": "Bob", "target": "Carol", "value": 43 }
+  //      ]
+  //    };
+  //
```

Comment:
Please remove all unused comments.

Reply:
This is a valuable comment, this is not a chunk of unused code. I'll tidy it up

Reply:
Oh, sorry. Didn't read the content.

Reply:
Simplified a little bit